content/docs/rfcs/17/README.md

```diff
@@ -182,6 +183,43 @@ Below is the description of the fields of `RateLimitProof` and their types.
 | `share_x` and `share_y`| array of 32 bytes each | Shamir secret shares of the user's secret identity key `sk` . `share_x` is the Poseidon hash of the `WakuMessage`'s `payload` concatenated with its `contentTopic` . `share_y` is calculated using [Shamir secret sharing scheme](/spec/32) | <!-- todo specify the poseidon hash setting -->
 | `nullifier`  | array of 32 bytes | internal nullifier derived from `epoch` and peer's `sk` as explained in [RLN construct](/spec/32)|
 
+## RLN Credentials format
+A RLN credential collects all the relevant fields that allow generation of valid RLN proofs with respect to an onchain membership contract.
+
+To make it compatible across different clients and platforms, RLN credentials will be encoded in JSON. 
+This further allows to validate credentials using [JSON schema](https://json-schema.org/).
+
+```
+{
+  "application": string,
+  "appIdentifier": string,
+  "credentials": [{
+    "key": string,
+    "commitment": string,
+    "membershipGroups" : [{
+      "chainId": number,
+      "contract": string,
+      "treeIndex": string
+    }]
+  }],
+  "version": number
+}
+```
+Fields are specified as follow:
+
+- `application` : the application name within which credential(s) refer to.
```

Comment:
Do we want to tie in the fqdn of the web app if relevant ? Similar to the `name` field in EIP-712?
Do we want to add

> Should this be possible, the usage of the credentials in a web app SHOULD be checked against the `application`, treating the string as the webapp's domain.